### PR TITLE
[Bug fix] binary_ng: disable PACK_RELU fast path for subtile-broadcast kernels

### LINKED ISSUE #44823: Elementwise op with activation having bad PCC if operand is implicitly broadcasted
Elementwise op with activations where one tensor is not the same shape as the other but is broadcastable produces incorrect results:
Following script can reproduce the problem:
```
import torch, ttnn

def pcc(x: torch.Tensor, y: torch.Tensor) -> float:
    x = x.to(torch.float32).flatten()
    y = y.to(torch.float32).flatten()
    cov = torch.mean((x - x.mean()) * (y - y.mean()))
    return float(cov / (x.std() * y.std() + 1e-12))

device = ttnn.open_device(device_id=0)
try:
    a = torch.randn(8, 64, 112, 112, dtype=torch.float32).reshape(1, 1, 8 * 112 * 112, 64)
    b = torch.randn(1, 1, 1, 64, dtype=torch.float32)
    ta = ttnn.from_torch(a, device=device, layout=ttnn.TILE_LAYOUT, dtype=ttnn.float32)
    tb = ttnn.from_torch(b, device=device, layout=ttnn.TILE_LAYOUT, dtype=ttnn.float32)
    golden = torch.relu(torch.add(a, b))

    out_fused = ttnn.to_torch(
        ttnn.add(
            ta,
            tb,
            activations=[ttnn.UnaryWithParam(ttnn.UnaryOpType.RELU)],
        )
    )
    print(f"pcc={pcc(out_fused, golden):.6f} ")
finally:
    ttnn.close_device(device)
```

Result:
`pcc=0.821169` 

If the tensors have the same shape, the output is correct.

### PR DESCRIPTION
**Pull request by Aswinmcw**
### Issue
Closes #44823

## Summary

When a fused post-activation is a single `RELU`, the program factory opts into a `PACK_RELU` optimization that applies `ZERO_RELU` via the packer config set once at the top of the compute kernel. This works for the no-broadcast case because no subsequent op touches the packer-relu state.

However, the subtile-broadcast kernels (`ROW` / `COL` / `SCALAR` / `ROW_COL`, both FPU and SFPU variants) do an intermediate `pack_tile(0, cb_llk_post)` followed by `pack_reconfig_data_format(cb_llk_post, cb_out)` per iteration, which clears the packer's `ZERO_RELU` state. By the time the final pack to `cb_out` runs, `PACK_RELU` is no longer active and **negative values pass through unclipped — the fused RELU is silently dropped**.

Other fused unary post-activations (GELU/SIGMOID/TANH/SILU) are unaffected because they go through the SFPU `PROCESS_POST_ACTIVATIONS` path, which is not coupled to the packer config.

### Fix

Restrict the `PACK_RELU` optimization to the non-broadcast case (`subtile_broadcast_type == NONE`) and let broadcast cases fall through to the SFPU activation path that is already proven correct for every other unary post-activation.

- No kernel changes.
- The `PACK_RELU` fast path is preserved for the common no-broadcast workload.
- Single point of change, less risk than re-arming packer-relu state inside ~8 broadcast compute kernels.

## Reproducer

FP32, Wormhole_B0:

```python
import torch, ttnn
device = ttnn.open_device(device_id=0)
a = torch.randn(8, 64, 112, 112).reshape(1, 1, 8 * 112 * 112, 64)
b = torch.randn(1, 1, 1, 64)            # row-broadcast in H
ta = ttnn.from_torch(a, device=device, layout=ttnn.TILE_LAYOUT, dtype=ttnn.float32)
tb = ttnn.from_torch(b, device=device, layout=ttnn.TILE_LAYOUT, dtype=ttnn.float32)
out = ttnn.to_torch(ttnn.add(ta, tb, activations=[ttnn.UnaryWithParam(ttnn.UnaryOpType.RELU)]))
golden = torch.relu(a + b)
# before: pcc ~ 0.82 (output equals (a + b) with no clipping)
# after : pcc = 1.000000, max_err = 0
```

## Verification

| case                                                      | before     | after          |
| ---                                                       | ---        | ---            |
| `add(a, b(1,1,1,64))` + fused RELU (`ROW_B`)              | pcc=0.821  | pcc=1.000000   |
| `add(a, b(1,1,H,1))` + fused RELU (`COL_B`)               | broken     | pcc=1.000000   |
| `add(a, b(1,1,1,1))` + fused RELU (`SCALAR`)              | broken     | pcc=1.000000   |
| `add(a, b_same_shape)` + fused RELU (no bcast)            | pcc=1.000  | pcc=1.000000 (PACK_RELU fast path intact) |
| Other fused acts (GELU / SIGMOID / TANH / SILU) on bcast  | OK         | OK             |

## Test plan

- [ ] CI passes on this branch
- [ ] No perf regression on no-broadcast eltwise (PACK_RELU optimization preserved)
- [ ] Existing fused-activation unit tests for binary_ng remain green
- [ ] Add regression unit test covering `ttnn.add` with broadcast shape + fused `RELU` activation for FLOAT32, BFLOAT16 (followup)

### CI Status
_Auto-generated on every push. Badges update live. Click a badge to filter runs by this branch._

- [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:Aswinmcw/fix-binary-ng-fused-relu-broadcast)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/runtime-sanity-tests.yaml?query=branch:Aswinmcw/fix-binary-ng-fused-relu-broadcast)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:Aswinmcw/fix-binary-ng-fused-relu-broadcast)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:Aswinmcw/fix-binary-ng-fused-relu-broadcast)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/all-model-tests.yaml?query=branch:Aswinmcw/fix-binary-ng-fused-relu-broadcast)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:Aswinmcw/fix-binary-ng-fused-relu-broadcast)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:Aswinmcw/fix-binary-ng-fused-relu-broadcast)
- [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:Aswinmcw/fix-binary-ng-fused-relu-broadcast)

### CODE REVIEW

**Review by Aswinmcw — COMMENTED** [review]
Clean, minimal fix. The root cause analysis is spot-on: `pack_reconfig_data_format(cb_llk_post, cb_out)` in the subtile-broadcast kernels clears the packer ZERO_RELU state, so the final pack to cb_out runs without RELU — negatives pass through silently.

The fix is the right call: single-line guard (`!is_subtile_broadcast`) that falls through to the SFPU activation path, which is already proven correct for all other unary post-activations. No kernel changes needed, PACK_RELU preserved for the common no-broadcast path.

Verification table covering ROW_B, COL_B, SCALAR, and no-bcast is thorough. The followup regression test (fused RELU + broadcast shapes for fp32/bf16) would be good to have.

**Comment by mouliraj-mcw** [comment]
**Request: land the regression test in this PR rather than as a followup.**

The PR description lists "Add regression unit test covering `ttnn.add` with broadcast shape + fused `RELU` activation for FLOAT32, BFLOAT16 (followup)". I checked `tests/ttnn/unit_tests/operations/eltwise/test_binary_bcast.py` — the only existing RELU + `ttnn.add` test (around line 4902–4911) uses same-shape inputs, so the broken cases — `ROW_B` / `COL_B` / `SCALAR` / `ROW_COL` × {FPU, SFPU} × {FP32, BF16, BF8_B, BF4_B} — are still not covered by CI after this fix.

Without a regression test in the same PR, the next refactor of the `kernels_ng/compute/*_bcast.cpp` kernels (e.g. re-enabling the PACK_RELU fast path, or re-arming `llk_pack_relu_config` after `pack_reconfig_data_format`) can silently reintroduce the bug — it would still pass PCC on the no-broadcast path, which is the only one currently exercised with fused RELU.

A minimal parametrization mirroring the verification table from the PR description would be sufficient:

- shapes: `(1,1,H,W)` vs `(1,1,1,W)` (ROW_B), `(1,1,H,1)` (COL_B), `(1,1,1,1)` (SCALAR), `(1,1,1,W)` vs `(1,1,H,1)` (ROW_COL)
- dtypes: at least `ttnn.float32` and `ttnn.bfloat16` (these hit different kernel variants via `is_llk_bcast` + `fp32_dest_acc_en`)
- `activations=[ttnn.UnaryWithParam(ttnn.UnaryOpType.RELU)]`
- golden: `torch.relu(a + b)` with `pcc >= 0.9999` (the bug manifests as pcc ≈ 0.82, so even a loose threshold catches it)

**Review by mateusznowakTT — APPROVED** [review]
